### PR TITLE
Rework messaging - Conversations/Group Messages

### PR DESCRIPTION
**Pull request by bruce-glazier**
### New feature changes

- Messages screen now displays Conversations instead of a list of user names
- New conversations can be added using the new Compose message screen
- Group messages are now supported

### Behavior changes for users
Some existing chat may not initially appear on the new message screen. A user can fix this by composing a new message to the other user they were previously chatting with. All prior dialog is preserved.

New service code should create conversations automatically users message each other so hopefully the above will be the exception, not the norm.

### Supporting changes

- Adds CacheProvider to automatically cache and dispose of values syncing to async-storage
- Adds UserProfilesProvider to fetch user profile details on sign-in and in the background. These are non-blocking queries that do not add to app launch loading time using placeholder data also leveraging the CacheProvider for efficiency. 

### Other fixes

- Adds isLoggedIn enabled flags to the react-query hooks useNotifications and the useActiveAccouns

## Next up
- Allow leaving conversations (automatically added back if someone messages in the thread)
- Fix sorting so unread is on top (service-side)

https://github.com/lifeomic/react-native-sdk/assets/76954025/eb83d7a1-86ce-4e5f-a565-aa0aee48147a

### CODE REVIEW

File: src/screens/MessageScreen.tsx [hunk 1]
```diff
@@ -116,24 +150,26 @@ export function MessageScreen({
...
-              right={() => renderRight(user.lastMessageTime)}
+              title={others
+                .filter((profile) => node.userIds.includes(profile.id))
+                .map((profile) => profile.profile.displayName)
+                .join(', ')}
+              description={`${
+                node.latestMessageUserId === userData?.id
+                  ? styles.selfPrefix
+                  : ''
+              }${node.latestMessageText}`}
```
Comment (sternetj) [inline src/screens/MessageScreen.tsx:171]: This should use i18n to combine the parts of the message. I also wonder if the prefix is necessary. Apple just displays the last message in the preview.

File: src/screens/ComposeMessageScreen.test.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+  );
+  const textInput = getByTestId('Type a message...');
+  fireEvent.changeText(textInput, 'some new message');
+
+  const sendButton = getByTestId('send-button');
+  fireEvent.press(sendButton);
+  expect(replaceMock).not.toHaveBeenCalled();
+});
+
+test('compose button disabled with message text but no user selected', async () => {
```
Comment (sternetj) [inline src/screens/ComposeMessageScreen.test.tsx:179]: Test title doesn't seem to match the test

**Review by sternetj — COMMENTED** [review]
Looks really solid. Had a few comments. Tests look awesome!

File: src/components/Messaging/ComposeInputBar.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    view: {
+      backgroundColor: theme.colors.elevation.level1,
+    },
+  } as ActivityIndicatorViewStyles,
+  inputToolbarContainer: {
+    position: 'absolute',
+    bottom: 10,
+    marginHorizontal: 16,
+  },
+  inputTextStyle: {},
```
[the remark this answers is not in the document]
Comment (bruce-glazier) [reply src/components/Messaging/ComposeInputBar.tsx:106]: Opps, probably not.

File: src/screens/ComposeMessageScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+}: HomeStackScreenProps<'Home/ComposeMessage'>) {
+  const { tileId } = route.params;
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const onChangeSearch = (query: string) => setSearchQuery(query);
+  const [selectedProfiles, setSelectedProfiles] = useState<User[]>([]);
+  const { styles } = useStyles(defaultStyles);
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      title: 'New Message',
```
[the remark this answers is not in the document]
Comment (bruce-glazier) [reply src/screens/ComposeMessageScreen.tsx:25]: 👍

File: src/screens/MessageScreen.tsx (hunk 1 above)
Comment (bruce-glazier) [reply src/screens/MessageScreen.tsx:171]: FWIW it was in our designs and I've enjoyed it from the perspective of testing but I'm certainly not set on it. If you think we should drop it, just say the word.

File: src/screens/ComposeMessageScreen.test.tsx (hunk 2 above)
Comment (bruce-glazier) [reply src/screens/ComposeMessageScreen.test.tsx:179]: Copy/paste, will fix

File: src/screens/MessageScreen.tsx (hunk 1 above)
Comment (bruce-glazier) [reply src/screens/MessageScreen.tsx]: Used i18n to combine the parts of a message but happy to remove the prefix if we think it isn't helpful.

Comment (sternetj) [reply src/screens/MessageScreen.tsx]: Let's keep it then!

**Review by sternetj — DISMISSED** [review]

**Review by sternetj — APPROVED** [review]

**Pull request merged**